Source/WebKit/WebProcess/WebPage/WebPage.cpp

```diff
@@ -6909,11 +6909,14 @@ void WebPage::firstRectForCharacterRangeAsync(const EditingRange& editingRange,
     if (!range)
         return completionHandler({ }, { });
 
-    // FIXME: Pass an EditingRange that matches the range of the first rect, rather than the entire passed-in range?
     auto rect = RefPtr(frame->view())->contentsToWindow(frame->editor().firstRectForRange(*range));
-    completionHandler(rect, editingRange);
-}
+    auto lineEndPosition = endOfLine(makeContainerOffsetPosition(range->start));
+    auto lineEndBoundary = makeBoundaryPoint(lineEndPosition);
+    if (!lineEndBoundary)
+        return completionHandler({ }, { });
 
+    completionHandler(rect, EditingRange::fromRange(*frame, makeSimpleRange(range->start, WTFMove(lineEndBoundary))));
```

Comment:
It would probably be a good idea to clamp this first line editing range to the given `editingRange`. I think we should also pass the given `editingRange` through, in the case where the selection is collapsed.